astropy/io/fits/file.py

```diff
@@ -462,6 +475,10 @@ def _try_read_compressed(self, obj_or_name, magic, mode, ext=''):
             if mode in ['update', 'append']:
                 raise OSError("update and append modes are not supported "
                               "with bzip2 files")
+            if not HAS_BZ2:
+                raise ValueError(
+                    ".bz2 format files are not supported since the Python "
+                    "interpreter does not include the bz2 module")
```

Comment:
I would prefer a simpler message along the lines of: `ImportError: Could not find built-in bz2 module.`

Reply:
I've just copied the exception verbatim from `get_readable_fileobj`, which is handling decompression for all other file types. But if it is too long, I would still keep both synchronised. And there is a point for making this an `ImportError`.
I am just a bit concerned a `Could not find ...` message might make the user think something with their path setup could be wrong rather than a genuine lack in their Python installation.

Reply:
Maybe `ImportError: This Python does not provide the bz2 module`?
Potentially does not even have to be the built-in one, if it is still possible to get the `bz2file` package to work under newer Python versions...

Reply:
Yeah, that sounds good.

Or since we only support Python 3.6+ we could even use [`ModuleNotFoundError`](https://docs.python.org/3/library/exceptions.html#ModuleNotFoundError)?